source/Editor.js

```diff
@@ -1953,16 +1954,16 @@ var keyHandlers = {
                     // Get the div containing the equation.
                     var parent = mathjaxParent.parentNode;
                     var nodesInParent = parent.childNodes;
-                    var secondLastNode = nodesInParent.childNodes[nodesInParent.length - 2];
+                    var secondLastNode = nodesInParent[nodesInParent.length - 2];
 
                     // Check if the equation is at the end of the node, the second last element. <br> is always the last element.
                     // If the equation is at the end of the node, set the caret before the equation, 
                     // otherwise set it at the end of the node, (the element before the <br> tag).
-                    if (secondLastNode && (secondLastNode === mathjaxParent)){
-                        var spanRange = document.createRange();
+                    var spanRange = document.createRange();
```

Comment:
Didn't we decide that this is the stile document, not the squire document? Shouldn't it be the squire document?